blocks/init/src/Blocks/components/heading/heading.php

```diff
@@ -36,6 +36,9 @@
 ?>
 
 <<?php echo esc_attr($headingLevel); ?> class="<?php echo esc_attr($headingClass); ?>" data-id="<?php echo esc_attr($unique); ?>">
+
 	<?php echo Components::outputCssVariables($attributes, $manifest, $unique, $globalManifest); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
+
 	<?php echo wp_kses_post($headingContent); ?>
+
```

Comment:
Why the empty spaces?

Reply:
Easier to spot what is going on

Reply:
Now that I look at it, this should be just one echo

```suggestion
	<?php
		echo Components::outputCssVariables($attributes, $manifest, $unique, $globalManifest),
		wp_kses_post($headingContent); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
	 ?>
```

Reply:
no don't do that please it looks strange and it is hard to see what is going on.  I will dismiss this suggestion

Reply:
Agree with Ivan, I think the current version looks much cleaner.

Reply:
Do as you wish, `echo` is a construct and has real-life performance benefits when used as intended (i.e. listing things to echo).

It's not just: oh we're doing this once. It's: heading is one of the most reused components so you're actually outputting this view template over and over and over. So it will add up in the end...

Reply:
Arguing for echo performance benefit? There's absolutely 0% chance this will EVER have ANY real-world performance drawbacks.

> Making something 'pretty' shouldn't really be an argument imo

Yes actually it does, while the above is nowhere near this situation: "easy to understand code" >>> "objectively faster / "better" code which is harder to read / understand".

The above reads more like HTML and HTML is the rest of this view. So it makes sense to make it fit and look more in place rather than having a negligible performance benefit, wouldn't you agree?

Reply:
> a negligible performance benefit, wouldn't you agree?

No, I won't agree, but we can agree to disagree 😉 

Personally, I don't find it hard to read since one is below another.